src/CaptureService/Capturer.cpp

```diff
@@ -1,94 +1,65 @@
-// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#include "CaptureService/CaptureService.h"
+#include "CaptureService/Capturer.h"
 
 #include <absl/time/time.h>
 
 #include "CaptureService/CommonProducerCaptureEventBuilders.h"
 #include "GrpcProtos/Constants.h"
-#include "GrpcProtos/capture.pb.h"
 #include "OrbitBase/Logging.h"
 #include "OrbitBase/Profiling.h"
 
 using orbit_grpc_protos::CaptureOptions;
-using orbit_grpc_protos::CaptureRequest;
-using orbit_grpc_protos::CaptureResponse;
 using orbit_grpc_protos::ProducerCaptureEvent;
 
-using orbit_producer_event_processor::GrpcClientCaptureEventCollector;
+using orbit_producer_event_processor::ClientCaptureEventCollector;
 using orbit_producer_event_processor::ProducerEventProcessor;
 
 namespace orbit_capture_service {
 
-CaptureService::CaptureService() {
+Capturer::Capturer() {
   // We want to estimate clock resolution once, not at the beginning of every capture.
   clock_resolution_ns_ = orbit_base::EstimateAndLogClockResolution();
 }
 
-void CaptureService::AddCaptureStartStopListener(CaptureStartStopListener* listener) {
+void Capturer::AddCaptureStartStopListener(CaptureStartStopListener* listener) {
   bool new_insertion = capture_start_stop_listeners_.insert(listener).second;
   ORBIT_CHECK(new_insertion);
 }
 
-void CaptureService::RemoveCaptureStartStopListener(CaptureStartStopListener* listener) {
+void Capturer::RemoveCaptureStartStopListener(CaptureStartStopListener* listener) {
   bool was_removed = capture_start_stop_listeners_.erase(listener) > 0;
   ORBIT_CHECK(was_removed);
 }
 
-grpc::Status CaptureService::InitializeCapture(
-    grpc::ServerReaderWriter<CaptureResponse, CaptureRequest>* reader_writer) {
+ErrorMessageOr<void> Capturer::InitializeCapture(
+    std::unique_ptr<ClientCaptureEventCollector> client_capture_event_collector) {
   {
     absl::MutexLock lock(&capture_mutex_);
     if (is_capturing_) {
-      return grpc::Status(grpc::StatusCode::ALREADY_EXISTS,
-                          "Cannot start capture because another capture is already in progress");
+      return ErrorMessage("Cannot start capture because another capture is already in progress");
     }
     is_capturing_ = true;
   }
 
-  grpc_client_capture_event_collector_ =
-      std::make_unique<GrpcClientCaptureEventCollector>(reader_writer);
-
-  producer_event_processor_ =
-      ProducerEventProcessor::Create(grpc_client_capture_event_collector_.get());
-
-  return grpc::Status::OK;
+  ORBIT_CHECK(client_capture_event_collector != nullptr);
+  client_capture_event_collector_.reset(client_capture_event_collector.release());
```

Comment:
Reading this line confused me quite a bit, because I was not spotting the `_`. Maybe use a `std::move`? To make it a bit more obvious?